x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/checksum/ChecksumBlobContainerIndexInput.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.index.store.checksum;
+
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.store.ByteBuffersDataOutput;
+import org.apache.lucene.store.ByteBuffersIndexOutput;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.elasticsearch.index.store.Store;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * A {@link IndexInput} that can only be used to verify footer checksums.
+ */
+public class ChecksumBlobContainerIndexInput extends IndexInput {
+
+    private final byte[] checksum;
+    private final long length;
+    private final long offset;
+
+    private int position;
+
+    private ChecksumBlobContainerIndexInput(String name, long length, byte[] checksum, IOContext context) {
+        super("ChecksumBlobContainerIndexInput(" + name + ')');
+        ensureReadOnceChecksumContext(context);
+        assert checksum.length == CodecUtil.footerLength();
+        this.checksum = Objects.requireNonNull(checksum);
+        this.length = length;
+        this.offset = length - this.checksum.length;
+        assert offset > 0;
```

Comment:
Should we set `position = -offset` here too? I think it's strange, although probably irrelevant, that a newly-opened index input has a nonzero file pointer.

Reply:
(that would imply that we'd need to throw appropriate exceptions if `position < 0` in `readByte()` and `readBytes()` too)

Reply:
Actually I think it's easier to reason about using a global `position` and adjust it only when reading the checksum bytes. I pushed 709e64424108b0863a96884cc682ad7bbae61d5a please let me know what you think